### PR TITLE
MB-10494-add-new-migration-to-index-users_roles-table

### PR DESCRIPTION
**Pull request by NamibiaTorres**
## Summary

Add a database migration to index the user_id and partially index deleted_at columns of the users_roles table.

## Setup to Run Your Code

<details>
<summary>Run office load testing in master and see how long it takes for issues in office user endpoints to arise. Then switch to my branch and make sure you have enough data in your database `make db_dev_e2e_populate`. Run load testing again in my branch. </summary>

##### Terminal 1

In the load testing repo run

```sh
make load_test_office
```

##### Terminal 2

Start the Go server locally in the mymove repo

```sh
make server_run
```

</details>

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: migrations/app/schema/20211202213138_index_user_roles_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- Add a database migration to index the user_id and deleted_at columns of the users_roles table.
+CREATE INDEX users_roles_user_id_idx ON users_roles(user_id);
+CREATE INDEX users_roles_not_deleted_partial_idx ON users_roles (deleted_at)
+    WHERE users_roles.deleted_at IS NULL;
+COMMENT ON INDEX users_roles_not_deleted_partial_idx IS 'indexes user_roles that are not deleted';
```
Comment (felipe-lee) [inline migrations/app/schema/20211202213138_index_user_roles_table.up.sql:5]:
```suggestion
COMMENT ON INDEX users_roles_not_deleted_partial_idx IS 'indexes users_roles that are not deleted';
```

File: migrations/app/schema/20211202213138_index_user_roles_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- Add a database migration to index the user_id and deleted_at columns of the users_roles table.
+CREATE INDEX users_roles_user_id_idx ON users_roles(user_id);
```
Comment (felipe-lee) [inline migrations/app/schema/20211202213138_index_user_roles_table.up.sql:2]: Small space change to make them be the same
```suggestion
CREATE INDEX users_roles_user_id_idx ON users_roles (user_id);
```

**Review by felipe-lee — APPROVED** [review]
Might be good to post in [#g-database](https://ustcdp3.slack.com/archives/CSGDM3NUW) to get a DB reviewer. 

The migration LGTM, I just small suggestions for spacing and a typo

File: migrations/app/schema/20211202213138_index_user_roles_table.up.sql (hunk 2 above)
Comment (NamibiaTorres) [reply migrations/app/schema/20211202213138_index_user_roles_table.up.sql:2]: Thanks Felipe! That table name gets me every time. I thought I made them all users_roles and missed one. When I make the changes, do I rerun the migrations? Or not?

Comment (felipe-lee) [reply migrations/app/schema/20211202213138_index_user_roles_table.up.sql:2]: No problem. You can locally if you want, though in this case it'll just change a comment line. They haven't run in the deployed environments so those are fine.

**Review by duncan-truss — APPROVED** [review]
Tested with load testing and looks like we're getting a slight improvement on create and login for office users 🎉

**Pull request merged**